Sources/Commands/Workspace.swift

```diff
@@ -269,6 +271,71 @@ public class Workspace {
         return path
     }
 
+    /// Updates the current dependencies.
+    public func updateDependencies() throws {
+        let (rootManifest, _) = try loadDependencyManifests()
+        // Only create constraints based on root manifest for the update resolution.
+        let updateConstraints = computePackageConstraints(rootManifest: rootManifest, dependencies: [])
+        // Resolve the dependencies.
+        let updateResults = try resolveDependencies(constraints: updateConstraints)
+
+        for (specifier, version) in updateResults {
```

Comment:
We should also remove any checkouts that are no longer used. I'd probably be tempted to write a loop the that iterated over the union of the new specifiers and the old specifiers, and for each specifier determine if it is ADDED, UPDATED, REMOVED, or UNCHANGED, which seem like the state changes a higher level client would want to report.